### ISSUE TITLE
smooth cam centering (on node center) is sometimes not smooth

### ISSUE THREAD

**Issue by xulman** [scenerygraphics/sciview#355]
suppose I select/activate a node that is displayed in the corner of the scene,
if I:

- double-clk the node in inspector, the node arrives smoothly to the scene's center (good)
- shift+mouseDrag, which should first bring the node to the scene centre and then start rotating the scene around it, the centering happens way too fast (feels like different code is centering it now, bad IMHO)

why is this not the same anymore?

**Comment by kephale** [comment]
Related to https://github.com/scenerygraphics/sciview/issues/319

**Comment by kephale** [comment]
I'll bump @skalarproduktraum on this one as well, since he wrote the smooth camera motion code.

**Comment by kephale** [comment]
Yeah, I checked this again, and it is kind of confusing for me to track what is going on with `AnimatedCenteringBeforeArcBallControl`

**Comment by xulman** [comment]
the patch is the following:
```diff
--- a/src/main/kotlin/sc/iview/Controls.kt
+++ b/src/main/kotlin/sc/iview/Controls.kt
@@ -240,13 +240,15 @@ open class Controls(val sciview: SciView) {
     private fun enableArcBallControl() {
         val h = inputHandler
 
-        val target: Vector3f = sciview.activeNode?.position ?: Vector3f(0.0f, 0.0f, 0.0f)
+        // please REVIEW (I'm no good at Kotlin):
+        // the original `position` is deprecated, was underlined in IDE
+        val target: Vector3f = sciview.activeNode?.spatialOrNull()?.position ?: Vector3f(0.0f, 0.0f, 0.0f)
 
         //setup ArcballCameraControl from scenery, register it with SciView's parameters
         val cameraSupplier = { sciview.currentScene.findObserver() }
-        val initAction = { _: Int, _: Int -> }
+        val initAction = { _: Int, _: Int -> sciview.centerOnActiveNode() } //DA FIX
         val scrollAction = { _: Double, _: Boolean, _: Int, _:Int -> }

         targetArcball = AnimatedCenteringBeforeArcBallControl(
                 initAction,
                 scrollAction,
```

this does the reasonably fast move to first make the active/selected obj in view centre and then runs the usual arcball on it; the issue was that the first part was skipped (scene jumped)

the initial movement itself may be cumbersome to some users, may be an optional tick in some future *ControlsPref*

**Comment by xulman** [comment]
I would have submitted PR right away, but I need

- help with the "please REVIEW" section
- opinion whether this `centerOnActiveNode()` should be optional (and guarded by some `Controls` flag)

**Comment by kephale** [comment]
@skalarproduktraum @moreApi see the above request for Kotlin feedback

**Comment by moreApi** [comment]
The kotlin part looks good. I can't say anytging about the other question.

**Comment by xulman** [comment]
> The kotlin part looks good. I can't say anytging about the other question.

Thank you @moreApi for the review

.... I was apparently lucky in my Kotlin syntax guessing ;-)
